### DISCUSSION TITLE
系统在生产环境部署时有多个网段即集群，此时apollo也会有多个cluster，如何防止配置不一致

### DISCUSSION THREAD

**Discussion by wangfan9002** [apolloconfig/apollo#3743] [Q&A]
系统在生产环境部署时有多个网段即集群，此时apollo也会有多个cluster，如何防止配置不一致； 
因为系统需要的配置参数比较多，人眼很难区分不同的apollo cluster有什么不同
有没有可能，以一个cluster作为baseA,  部署系统的一个网段再指定另一个clusterB, 这个clusterB中只配置特殊的不一样的配置

**Comment by Anilople** [comment]
现在不支持

环境+集群+namespace确定一份配置

---

你的需求是

DEV+**default**+application 配置公共的配置

然后
DEV+**beijing**+application 配置特殊的配置，同时也使用DEV+**default**+application作为基础配置

DEV+**shanghai**+application 配置特殊的配置，同时也使用DEV+**default**+application作为基础配置

这样 DEV+**beijing**+application 和 DEV+**shanghai**+application 就可以不用自己维护公共的配置了，是这样吗？

目前是不支持这样做的

**Reply by wangfan9002** [reply]
恩，是这个意思，是希望 default 里是公用配置， 有100个key,   shanghai 只有几个key， beijing有几个key,  beijing，shanghai的这几个key优先级更高一点，能覆盖default里同名的key；  这样， 我一眼就能看出来，其他 cluster(beijing, shanghai) 里有哪些不同的配置

**Reply by nobodyiam** [reply]
公共的 namespace 应该可以满足这个场景，比如在某个公共项目中配置 DEV+**default**+somePublicNamespace 存放公共配置
然后
在你应用的 DEV+**beijing**+somePublicNamespace 配置特殊的配置，同时也会使用公共项目的 DEV+**default**+somePublicNamespace作为基础配置

在你应用的 DEV+**shanghai**+somePublicNamespace 配置特殊的配置，也会使用公共项目的 DEV+**default**+somePublicNamespace作为基础配置
